plugins/inventory/now.py

```diff
@@ -605,6 +653,16 @@ def parse(self, inventory, loader, path, cache=True):
             client = Client(**self._get_instance())
         except ServiceNowError as e:
             raise AnsibleParserError(e)
+
+        enhanced = self.get_option("enhanced")
+
+        if enhanced and (named_groups or group_by):
+            self.display.warning(
```

Comment:
I think we should fail instead of proceeding with a warning.

cc @tima @jillr what do you think?